src/sentry/api/event_search.py

```diff
@@ -1365,233 +1365,354 @@ def get_default(self, params):
+
+        # use default values to populate optional arguments if any
+        for argument in self.args[len(columns) :]:
+            try:
+                default = argument.get_default(params)
+            except InvalidFunctionArgument as e:
+                raise InvalidSearchQuery(u"{}: invalid arguments: {}".format(field, e))
+
+            # Hacky, but we expect column arguments to be strings so easiest to convert it back
+            columns.append(six.text_type(default) if default else default)
+
+        arguments = {}
+
+        # normalize the arguments before putting it in a dict
+        for column_value, argument in zip(columns, self.args):
+            try:
+                normalized_value = argument.normalize(column_value)
+                arguments[argument.name] = normalized_value
+            except InvalidFunctionArgument as e:
+                raise InvalidSearchQuery(
+                    u"{}: {} argument invalid: {}".format(field, argument.name, e)
+                )
+
+        # populate any computed args
+        for calculation in self.calculated_args:
+            arguments[calculation["name"]] = calculation["fn"](arguments)
+
+        return arguments
+
+    def validate(self):
+        # assert that all optional args have defaults available
+        for i, arg in enumerate(self.optional_args):
+            assert (
+                arg.has_default
+            ), u"{}: optional argument at index {} does not have default".format(self.name, i)
+
+        # assert that the function has only one of the following specified
+        # `column`, `aggregate`, or `transform`
+        assert (
+            sum([self.column is not None, self.aggregate is not None, self.transform is not None])
+            == 1
+        ), u"{}: only one of column, aggregate, or transform is allowed".format(self.name)
+
+    def validate_argument_count(self, field, arguments):
+        """
+        Validate the number of required arguments the function defines against
+        provided arguments. Raise an exception if there is a mismatch in the
+        number of arguments. Do not return any values.
+
+        There are 4 cases:
+            1. provided # of arguments != required # of arguments AND provided # of arguments != total # of arguments (bad, raise an error)
+            2. provided # of arguments < required # of arguments (bad, raise an error)
+            3. provided # of arguments > total # of arguments (bad, raise an error)
+            4. required # of arguments <= provided # of arguments <= total # of arguments (good, pass the validation)
+        """
+        args_count = len(arguments)
+        total_args_count = self.total_args_count
+        if args_count != total_args_count:
+            required_args_count = self.required_args_count
+            if required_args_count == total_args_count:
+                raise InvalidSearchQuery(
+                    u"{}: expected {:g} arguments".format(field, total_args_count)
```

Comment:
I think you still need to handle the "1 arguments" case here.

Reply:
I think you were looking at an older commit? I don't see this anymore.

Reply:
My bad, ignore.